instrumentor/controllers/instrumentationdevice/pods_webhook.go

```diff
@@ -28,7 +32,58 @@ func (p *PodsWebhook) Default(ctx context.Context, obj runtime.Object) error {
 		pod.Annotations = map[string]string{}
 	}
 
-	//pod.Annotations["odigos.io/instrumented-webhook"] = "true"
-	log.V(0).Info("Defaulted Pod", "name", pod.Name)
+	// Inject ODIGOS environment variables into all containers
+	injectOdigosEnvVars(pod)
+
 	return nil
 }
+
+func injectOdigosEnvVars(pod *corev1.Pod) {
```

Comment:
@tamirdavid1 Because we are adding the mutating webhook with `reinvocationPolicy: IfNeeded` Kubernetes may call us multiple times in case other webhooks are also modifying the pod (this config guarantee that we are the last webhook called but we may be called multiple times)
In that case I think we will add these env vars multiple times, maybe we should check and add these env vars only if they are not already exists in the pod object